src/rsa.rs

```diff
@@ -163,8 +165,107 @@ fn parse_public_key<'a>(input: untrusted::Input<'a>) ->
     })
 }
 
+#[allow(non_camel_case_types)]
+enum BIGNUM {}
+#[allow(non_camel_case_types)]
+enum BN_MONT_CTX {}
+
+#[allow(non_camel_case_types)]
+#[repr(C)]
+struct RSA {
+    n: *mut BIGNUM,
+    e: *mut BIGNUM,
+    d: *mut BIGNUM,
+    p: *mut BIGNUM,
+    q: *mut BIGNUM,
+    dmp1: *mut BIGNUM,
+    dmq1: *mut BIGNUM,
+    iqmp: *mut BIGNUM,
+    mont_n: *mut BN_MONT_CTX,
+    mont_p: *mut BN_MONT_CTX,
+    mont_q: *mut BN_MONT_CTX,
+    mont_qq: *mut BN_MONT_CTX,
+    qmn_mont: *mut BIGNUM,
+    iqmp_mont: *mut BIGNUM,
+}
+
+/// An RSA key pair.
+pub struct RSAKeyPair {
+    rsa: Box<RSA>,
+}
+
+const BN_FLG_CONSTTIME: c::int = 4;
+
+fn build_bn(input: untrusted::Input) -> *mut BIGNUM {
+    unsafe {
+        let bytes = input.as_slice_less_safe();
+        let res = BN_bin2bn(bytes.as_ptr(), bytes.len(), ptr::null_mut());
+        BN_set_flags(res, BN_FLG_CONSTTIME);
+        res
+    }
+}
+
+impl RSAKeyPair {
+    /// Reads key data from the given DER-encoded input, which must contain
+    /// the full private key data (n, e, d, p, q, dmp1, dmq1, iqmp).
```

Comment:
Please base the documentation of this function on the documentation for other, similar things. For example, see https://briansmith.org/rustdoc/ring/signature/static.ECDSA_P384_SHA256_VERIFY.html. Notice that it references the specification, in this case [RFC3447 Appendix A.1.2](https://tools.ietf.org/html/rfc3447#appendix-A.1.2). Instead of mentioning (`n`, `e`, `d`, ...) individually, just reference the `RSAPrivateKey` structure.

Please mention that only two-prime keys (version 0) are supported; multiprime keys (version 1) are not yet supported.